[botan] Cross compiling for x86_64 on Apple Silicon fails

### Operating system

macOS

### Compiler

Apple-Clang

### Steps to reproduce the behavior

```Shell
On an Apple Silicon (arm64) machine:
`./vcpkg install botan:x64-osx`
```


### Failure logs

  ```
   INFO: /Users/ruurd/Downloads/vcpkg/buildtrees/botan/src/2.19.3-3b622dd437.clean/configure.py invoked with options "--distribution-info=vcpkg x64-osx --with-pkg-config --link-method=copy --with-debug-info --includedir=include --bindir=bin --libdir=lib --without-documentation --with-external-includedir=/Users/ruurd/Downloads/vcpkg/installed/x64-osx/include --cpu=x86_64 --cc=clang --disable-shared --enable-static --debug-mode --prefix=/Users/ruurd/Downloads/vcpkg/packages/botan_x64-osx/debug --with-external-libdir=/Users/ruurd/Downloads/vcpkg/installed/x64-osx/debug/lib"
   INFO: Configuring to build Botan 2.19.3 (revision unknown)
   INFO: Running under 3.11.3 (v3.11.3:f3909b8bc8, Apr  4 2023, 20:12:10) [Clang 13.0.0 (clang-1300.0.29.30)]
   INFO: CXX environment variable is set which will override compiler path
   INFO: Implicit --ar-command=/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ar due to environment variable AR
   INFO: Implicit --cc-bin=/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/c++ due to environment variable CXX
   INFO: Implicit --cxxflags=-arch x86_64 -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX13.3.sdk -fPIC -g due to environment variable CXXFLAGS
   INFO: Implicit --ldflags=-arch x86_64 -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX13.3.sdk due to environment variable LDFLAGS
   INFO: Autodetected platform information: OS="Darwin" machine="arm64" proc="arm"
   INFO: Guessing target OS is darwin (use --os to set)
   INFO: Using /etc/ssl/cert.pem as system certificate store
   INFO: Auto-detected compiler version 4.0
   INFO: Auto-detected compiler arch arm64
  ERROR: Configured target is x86_64 but compiler probe indicates arm64
```

```
Computing installation plan...
The following packages will be built and installed:
    botan[core]:x64-osx -> 2.19.3
Detecting compiler hash for triplet x64-osx...
Restored 0 package(s) from /Users/ruurd/.cache/vcpkg/archives in 5.38 us. Use --debug to see more details.
Installing 1/1 botan:x64-osx...
Building botan[core]:x64-osx...
-- Using cached randombit-botan-2.19.3.tar.gz.
-- Cleaning sources at /Users/ruurd/Downloads/vcpkg/buildtrees/botan/src/2.19.3-3b622dd437.clean. Use --editable to skip cleaning for the packages you specify.
-- Extracting source /Users/ruurd/Downloads/vcpkg/downloads/randombit-botan-2.19.3.tar.gz
-- Applying patch fix-generate-build-path.patch
-- Applying patch embed-debug-info.patch
-- Applying patch arm64-windows.patch
-- Applying patch pkgconfig.patch
-- Applying patch verbose-install.patch
-- Applying patch configure-zlib.patch
-- Applying patch fix-objectfile-list.patch
-- Using source at /Users/ruurd/Downloads/vcpkg/buildtrees/botan/src/2.19.3-3b622dd437.clean
-- Found external ninja('1.11.1').
-- Getting CMake variables for x64-osx
-- Getting CMake variables for x64-osx-dbg
-- Getting CMake variables for x64-osx-rel
-- Configuring x64-osx-dbg
CMake Error at scripts/cmake/vcpkg_execute_required_process.cmake:112 (message):
    Command failed: /bin/bash -c "V=1 ./../src/2.19.3-3b622dd437.clean/configure  \"/Library/Frameworks/Python.framework/Versions/3.11/bin/python3\" \"/Users/ruurd/Downloads/vcpkg/buildtrees/botan/src/2.19.3-3b622dd437.clean/configure.py\" \"--distribution-info=vcpkg x64-osx\" \"--with-pkg-config\" \"--link-method=copy\" \"--with-debug-info\" \"--includedir=include\" \"--bindir=bin\" \"--libdir=lib\" \"--without-documentation\" \"--with-external-includedir=/Users/ruurd/Downloads/vcpkg/installed/x64-osx/include\" \"--cpu=x86_64\" \"--cc=clang\" \"--disable-shared\" \"--enable-static\" \"--debug-mode\" \"--prefix=/Users/ruurd/Downloads/vcpkg/packages/botan_x64-osx/debug\" \"--with-external-libdir=/Users/ruurd/Downloads/vcpkg/installed/x64-osx/debug/lib\" \"--prefix=/Users/ruurd/Downloads/vcpkg/installed/x64-osx/debug\""
    Working Directory: /Users/ruurd/Downloads/vcpkg/buildtrees/botan/x64-osx-dbg
    Error code: 1
    See logs for more information:
      /Users/ruurd/Downloads/vcpkg/buildtrees/botan/config-x64-osx-dbg-out.log
      /Users/ruurd/Downloads/vcpkg/buildtrees/botan/config-x64-osx-dbg-err.log

Call Stack (most recent call first):
  scripts/cmake/vcpkg_configure_make.cmake:833 (vcpkg_execute_required_process)
  ports/botan/portfile.cmake:125 (vcpkg_configure_make)
  scripts/ports.cmake:147 (include)


error: building botan:x64-osx failed with: BUILD_FAILED
```

### Additional context

The problem seems to be that Botan auto-discovers the compiler to be arm64:

> ERROR: Configured target is x86_64 but compiler probe indicates arm64

--- a/ports/botan/portfile.cmake
+++ b/ports/botan/portfile.cmake
@@ -32,6 +32,7 @@ endif()
 
 vcpkg_list(SET configure_arguments
     "--distribution-info=vcpkg ${TARGET_TRIPLET}"
+    --disable-cc-tests
     --with-pkg-config
     --link-method=copy
     --with-debug-info

--- a/ports/botan/vcpkg.json
+++ b/ports/botan/vcpkg.json
@@ -1,6 +1,7 @@
 {
   "name": "botan",
   "version": "2.19.3",
+  "port-version": 1,
   "description": "A cryptography library written in C++11",
   "homepage": "https://botan.randombit.net",
   "license": "BSD-2-Clause",

--- a/versions/b-/botan.json
+++ b/versions/b-/botan.json
@@ -1,6 +1,11 @@
 {
   "versions": [
     {
+      "git-tree": "7b9240ee18f12e8acaedf11613aa6bc1d7aad9d5",
+      "version": "2.19.3",
+      "port-version": 1
+    },
+    {
       "git-tree": "1ad976f3e650d7188414a0c2c62ec7290bed64e8",
       "version": "2.19.3",
       "port-version": 0

--- a/versions/baseline.json
+++ b/versions/baseline.json
@@ -1238,7 +1238,7 @@
     },
     "botan": {
       "baseline": "2.19.3",
-      "port-version": 0
+      "port-version": 1
     },
     "box2d": {
       "baseline": "2.4.1",

